prow/clonerefs/run.go

```diff
@@ -81,7 +82,24 @@ func (o Options) Run() error {
 		go func() {
 			defer wg.Done()
 			for ref := range input {
-				output <- cloneFunc(ref, o.SrcRoot, o.GitUserName, o.GitUserEmail, o.CookiePath, env, oauthToken)
+				// Attempt to clone the repo
+				// If the cluster is autoscaling, it seems Prow can hit DNS too hard on new nodes
+				// So we retry if have a failure we can guess is due to DNS
+				// Exponential backoff is targeted for a little over 5 minutes on the last try
+				for i := 1; i < 4; i++ {
+					record := cloneFunc(ref, o.SrcRoot, o.GitUserName, o.GitUserEmail, o.CookiePath, env, oauthToken)
+					if record.Failed {
+						failed := record.Commands[len(record.Commands)-1]
+						if strings.Contains(failed.Error, "Could not resolve host") {
+							seconds := time.Second * 25 * i * i
+							logrus.Infof("DNS failure detected; retrying clone in %d seconds", seconds)
+							time.Sleep(seconds)
```

Comment:
We should not sleep when `i == 3`

Reply:
Why not? That's the intention.

Reply:
oh, I see; I'm doing the sleep wrong. Oh darn.

Reply:
It failed all four times and will now exit. There's no need to make it sleep for 400 seconds before reporting this failure.